perceval/archive.py

```diff
@@ -342,3 +343,123 @@ def _count_table_rows(self, table_name):
             cursor.close()
 
         return row[0]
+
+
+class ArchiveManager:
+    """Manager for handling archives in Perceval.
+
+    This class manages the creation, deletion and access of `Archive`
+    objects. Archives are stored under `dirpath` directory, using
+    a random SHA1 for each file. The first byte of the hashcode will
+    be the name of the subdirectory; the remaining bytes, the archive
+    name.
+
+    :param: dirpath: path where the archives are stored
+    """
+
+    STORAGE_EXT = '.sqlite3'
+
+    def __init__(self, dirpath):
+        self.dirpath = dirpath
+
+        if not os.path.exists(self.dirpath):
+            os.makedirs(self.dirpath)
+
+    def create_archive(self):
+        """Create a new archive.
+
+        The method creates in the filesystem a brand new archive with
+        a random SHA1 as its name. The first byte of the hashcode will
+        be the name of the subdirectory; the remaining bytes, the
+        archive name.
+
+        :returns: a new `Archive` object
+
+        :raises ArchiveManagerError: when an error occurs creating the
+            new archive
+        """
+        hashcode = uuid.uuid4().hex
+        archive_dir = os.path.join(self.dirpath, hashcode[0:2])
+        archive_name = hashcode[2:] + self.STORAGE_EXT
+        archive_path = os.path.join(archive_dir, archive_name)
+
+        if not os.path.exists(archive_dir):
+            os.makedirs(archive_dir)
+
+        try:
+            archive = Archive.create(archive_path)
+        except ArchiveError as e:
+            raise ArchiveManagerError(cause=str(e))
+
+        return archive
+
+    def remove_archive(self, archive_path):
+        """Remove an archive.
+
+        This method deletes from the filesystem the archive stored
+        in `archive_path`.
+
+        :param archive_path: path to the archive
+
+        :raises ArchiveManangerError: when an error occurs removing the
+            archive
+        """
+        try:
+            Archive(archive_path)
+        except ArchiveError as e:
+            raise ArchiveManagerError(cause=str(e))
+
+        os.remove(archive_path)
```

Comment:
Hmmm, but this just removes a file (archive_path must be a path to a file). I have supposed that in the Archive there will be several files. Am I missing something?

Reply:
Looking to the code below, it seems that for the same data source, several Archive objects exists. So the relation is one file per each Archive object. And the ArchiveManager is the one managing all those Archive objects for a specific configuration: origin, backend, category. Right?

Reply:
Each time `perceval` runs, it creates a `Archive` object/file for that process. All the items fetched on that execution are stored in that archive. `ArchiveManager` manages all those `Archive` objects, no matter which the origin or backend were.